### PR TITLE
Improve the verication of member field

### LINKED ISSUE #330: Client form: improve fields validation
### Expected Behaviour

The following fields should be properly validated:
- Postal code
- Phone number
- Email
### Actual Behaviour

No validation (only required)

### PR DESCRIPTION
**Pull request by fpeyre**
### Changes proposed in this pull request:
- Use forms.EmailField to improve email verification in the form.
- Use the library django-localflavor who give CAPhoneNumberField, CAPostalCodeField to verify the good format of a phone number and postal code number (more information here : https://django-localflavor.readthedocs.io/en/latest/localflavor/ca/ ). CAPhoneNumberField display the value with the format we desired in issue #491 
- Overload print function for phone number. CAPhoneNumberField register the number in the good format, but we can have inside the databse number with the old format. They must be print in the rigth format inside our templates
### Status
- [X] READY
### How to verify this change
- Test a postal code field. He must respect the canadian postal code synthax
- Test a phone field with and without extension (#). The phone can be wrote as XXXXXXXXXXX, XXX XXX XXXX or XXX-XXX-XXXX
-  The phone number is displayed in the same manner for each client (XXX-XXX-XXXX) with its extension if needed
### Deployment notes and migration
- [X]  Migration is needed for this change

### Additional notes
We must rewrite the clean method of CAPhoneNumberField to include phone number with extensions
We must overload the print function of contact and member field in link to a phone number to correctly display it inside our template

### CODE REVIEW

**Comment by manumilou** [comment]
Needs review guys!

**Comment by manumilou** [comment]
Pretty good @fpeyre . 
Can you add a formatter even if the phone number is not well formatted in the database ?

File: src/member/models.py [hunk 1]
```diff
@@ -151,6 +151,9 @@ def add_contact_information(self, type, value, force_update=False):
         updated. Otherwise, it should create a new one.
         """
         created = False
+        if value is None:
+            value = ''
+
         if force_update or value is not '':
             contact, created = Contact.objects.update_or_create(
```
Comment (erozqba) [inline src/member/models.py]: @fpeyre, question here: Why not use `if force_update or value is not None:` instead of adding the if condition?

Comment (fpeyre) [inline src/member/models.py]: Hmm it is to fix test who broke with error message indicate we try to insert NULL value for the field 'value' inside the database.
But sometimes we can have force_value = true with value=None, who raise the exception. So i consider None as an empty string to not raise the error.

**Comment by fpeyre** [comment]
@manumilou I add a formater even if the phone number is not well formatted in the database . You can review it.

**Comment by manumilou** [comment]
@fpeyre Please limit your commit message to 80 characters. I'll review your code while you are doing that :-)

**Pull request merged**